docs/data-structures/binary-tree.mdx

```diff
@@ -0,0 +1,792 @@
+        } 
+    }
+    void inorder_rec(){                  
+        inorder_rec(root);
+    }
+    void inorder_rec(treenode *temp){       //recursive inorder   
+        if(temp!=NULL){
+            inorder_rec(temp->left);
+            cout<<temp->data<<" ";
+            inorder_rec(temp->right);
+        }
+    }
+    void inorder_nonrec(){          //non recursive using stack
+        treenode *temp;
+        stack<treenode *> s;
+        temp = root;
+        while(1){
+            while(temp!=NULL){                //firstly travesing left of tree
+                s.push(temp);
+                temp=temp->left;
+            }
+            if(s.empty()){
+                break;
+            }
+            temp = s.top();
+            s.pop();
+            cout<<temp->data<<" ";
+            temp=temp->right; 
+        }
+        cout<<endl;
+    }
+};
+
+int main(){
+    tree t;
+    t.create();
+    t.inorder_rec();
+    cout<<endl;
+    t.inorder_nonrec();
+}
+```
+
+</TabItem>
+
+<TabItem value="py">
+
+```python
+# InOrder tree traversal using python3 (Iterative)
+
+class node:
+    def __init__(self, data):
+        self.data = data
+        self.left = None
+        self.right = None
+
+class Tree:
+    def __init__(self):
+        self.root = None
+
+    # creation of tree.
+    def createTree(self, NodesArray):
+        queue = []
+        self.root = node(int(NodesArray.pop(0)))
+        queue.append(self.root)
+        size = len(queue)
+        while NodesArray:
+            while size:
+                root = queue.pop(0)
+                if root:
+                    data = NodesArray.pop(0)
+                    if data == 'null':
+                        root.left = None
+                    else:
+                        root.left = node(int(data))
+                        queue.append(root.left)
+                    data = NodesArray.pop(0)
+                    if data == 'null':
+                        root.right = None
+                    else:
+                        root.right = node(int(data))
+                        queue.append(root.right)
+                size -= 1
+            size = len(queue)
+
+    # Iterative function to perform in-order traversal of the tree
+    def inorderIterative(self,root):
+	    stack = []
+	    curr = root
+	    while stack or curr:
+        # if current node is not null, push it to the stack
+		    if curr:
+		    	stack.append(curr)
+		    	curr = curr.left
+		    else:
+			# else if current node is null, we pop an element from the stack,print it and finally set current node to its right child
+			    curr = stack.pop()
+			    print(curr.data, end=' ')
+			    curr = curr.right
+print("Enter the input in the sequence :- root.val root->left root->right (enter null if empty)")
+NodesArray = list(input().split(' '))
+InOrdertree = Tree()
+```
+
+</TabItem>
+</Tabs>
+
+### Post-Order Traversal:
+
+<Tabs
+  defaultValue="py"
+  values={[
+    { label: 'Python', value: 'py', },
```

Comment:
Why can't we use C++ here?

Reply:
> Why can't we use C++ here?
 I couldn't find C++ code for this traversal. So I used java and python. Shall I put java here?